src/components/dialogs/settingsDialog/DownloadSettings.vue

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable vue/max-attributes-per-line */
+<template>
+  <v-expansion-panel>
+    <v-expansion-panel-header> Downloads </v-expansion-panel-header>
+    <v-expansion-panel-content>
+      <h4>When adding a torrent</h4>
+      <v-divider />
+
+      <v-container
+        class="px-0"
+        fluid
+      >
+        <v-switch
+          :input-value="preferences.create_subfolder_enabled"
+          :label="`Create subfolder for torrents with multiple files`"
+          @change="changeSettings('create_subfolder_enabled', !preferences.create_subfolder_enabled)"
+        />
+      </v-container>
+    </v-expansion-panel-content>
+  </v-expansion-panel>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import { Preferences } from "@/types";
+import { Component } from "vue-property-decorator";
+import { mapActions, mapGetters } from "vuex";
+
+@Component({
+  components: {},
+  computed: {
+    ...mapGetters({
+      preferences: "allPreferences",
+    }),
+  },
+  methods: {
+    ...mapActions({
+      updatePreferencesRequest: 'updatePreferencesRequest'
+    })
+  },
+})
+export default class DownloadSettings extends Vue {
+  preferences!: Preferences;
+
+  updatePreferencesRequest!: (_: any) => void;
+
+  changeSettings(property: string, value: string | boolean) {
+    // copy of preferences with changed certrain property and its value
+    const updatedPreferences = Object.assign({}, this.preferences, {
```

Comment:
It's possible to just post changed preference.

Reply:
I am not sure what are you suggesting. Can you please explain?

Reply:
Please see note 1 in https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)#set-application-preferences

Reply:
Got it!